packages/aws-cdk/lib/api/deploy-stack.ts

```diff
@@ -181,7 +185,16 @@ export interface DeployStackOptions {
    *
    * @default true
    */
-  readonly rollback?: boolean;
+  rollback?: boolean;
```

Comment:
I don't like changing this to not be `readonly`. See my solution below on how to handle this in a more idiomatic way. This change should be reverted.